packages/core/test/globals.spec.js

```diff
@@ -317,6 +317,60 @@ test('globals - nested property true.false.true', async (t) => {
   t.is(testResult.a_b_notOk, false)
 })
 
+test('globals - window-like accessors taming', async (t) => {
+  'use strict'
+  const shared = {
+    context: Object.defineProperties(
+      Object.create(null), {
+        top: {get: () => ({})},
+        window: {get: () => ({})},
+        frames: {get: () => ({})},
+        parent: {get: () => ({})},
+        self: {get: () => ({})},
+      }),
+    config: {
+      resources: {
+        one: {
+          globals: {
+            top: true,
+            window: true,
+            frames: true,
+            parent: true,
+            globalThis: true,
+            self: true,
+            "console.warn": true,
+          },
+        },
+      },
+    },
+  }
+  const handlesAccess = createScenarioFromScaffold({
+    defineOne: () => {
+      module.exports = {
+        top: globalThis === globalThis.top,
+        window: globalThis === globalThis.window,
+        frames: globalThis === globalThis.frames,
+        parent: globalThis === globalThis.parent,
+        globalThis: globalThis === globalThis.globalThis,
+        self: globalThis === globalThis.self,
+        warn: typeof globalThis.top.window.frames.parent.self.console.warn,
+        URL: typeof globalThis.top.window.frames.parent.self.URL
+      }
+    },
+    ...shared,
+  })
+
+  const testResult = await runScenario({ scenario: handlesAccess })
+  t.is(testResult.top, true)
```

Comment:
This could be asserted all at once with a deepEqual and would produce more useful output when failing.

Reply:
59f5c3f3e